comm/src/main/scala/coop/rchain/comm/unicast.scala

```diff
@@ -3,6 +3,11 @@ package coop.rchain.comm
 import java.net.{DatagramPacket, DatagramSocket}
 import scala.util.{Failure, Success, Try}
 
+sealed trait DatagramError extends Throwable
```

Comment:
I'm not sure that this is a good practice to inherit from `Throwable`. In my practice, I am used constructors with exceptions, like:
```scala
case class DatagramOverflow(size: Int, ex: Exception) extends DatagramError
```

Reply:
It has to end up as a `Throwable` for the `Try[...]` later on. I can change that, if it's bad practice, but that'll be a separate PR. Let me know when you see it all assembled.

Reply:
You would use `Either[DatagramError, Foo]` instead of `Try[Foo]`. So no need to extend from `Throwable`.

Reply:
I guess I don't follow. In order to use `Either[DatagramError, Foo]` instead of `Try[foo]` after the `encode`/`decode` business, I'll have to catch all the exceptions and turn them into `DatagramError`s anyway. I'm willing to do that, but isn't that the whole point of `Try[Foo]` in the first place?

Reply:
`Try(...)` turns your computation into a `Success` or a `Failure` while with `Either` you can further specify the type of the error.

Reply:
I will issue a separate PR that uses `Either` in favor of `Try` for return values. It will included an expanded set of `DatagramError` subtypes.